### PR TITLE
Ergonomic TCI lib

### LINKED ISSUE #1178: [ETHEREUM-CONTRACTS] New Version of SuperTokenLibrary
# Key Issues

- Some updated understanding of the roles and responsibilities of cfa/ida libraries and forwarders.
- Fix https://github.com/superfluid-finance/protocol-monorepo/issues/1153.

### PR DESCRIPTION
**Pull request by d10r**

### CODE REVIEW

File: packages/ethereum-contracts/contracts/apps/TCILibraryV1.sol [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+        uint8 permissionsBitmask;
+        (, permissionsBitmask, flowRateAllowance) = cfa.getFlowOperatorData(token, sender, flowOperator);
+        allowCreate = permissionsBitmask & 1 == 1 ? true : false;
+        allowUpdate = permissionsBitmask >> 1 & 1 == 1 ? true : false;
+        allowDelete = permissionsBitmask >> 2 & 1 == 1 ? true : false;
+    }
+
+    // ************** helpers **************
+
+    // keccak256("Superfluid.v1")
```
Comment (hellwolf) [inline packages/ethereum-contracts/contracts/apps/TCILibraryV1.sol]: Looks good, just to give a bit more thinking re storage slot conflict possibility considering [solidity storage state layout rules](https://docs.soliditylang.org/en/latest/internals/layout_in_storage.html)

Make the hashed string also a bit more specific (java path style, e.g.)

Comment (saflamini) [reply packages/ethereum-contracts/contracts/apps/TCILibraryV1.sol]: Makes sens re: @hellwolf

**Comment by saflamini** [comment]
@d10r just pushed my recent changes - want to add back your cosmetic changes we discussed today?

File: packages/ethereum-contracts/scripts/deploy-aux-contracts.d.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+declare const _exports: any;
```
Comment (hellwolf) [inline packages/ethereum-contracts/scripts/deploy-aux-contracts.d.ts]: 1). scripts -> ops-scripts
2). ewww

**Comment by hellwolf** [comment]
Please mark it as ready when so, then one more review with the team we are good to go.

**Comment by saflamini** [comment]
cc @d10r - @sunnyjaycer and I just gave it one final review and feel good about where we're at.

I will begin prepping some documentation for this as we get ready to give final review

**Comment by hellwolf** [comment]
check Changelog Reminder

@d10r

**Review by hellwolf — APPROVED** [review]

**Pull request merged**